api/src/test/java/org/openmrs/module/openconceptlab/client/OclClientTest.java

```diff
@@ -132,4 +133,37 @@ public void parseDateFromPath_shouldParseDate() throws Exception {
 		
 		assertThat(date, is(calendar.getTime()));
 	}
+
+	@Test
+	public void getUrlVersion_shouldGetUrlVersion() throws IOException {
+		String url = "https://api.openconceptlab.org/users/username/collections/collectionname/v1.0";
+		String subscriptionToken = "53fc72f0498a707a26e4d903c0f24c2db24d1e35";
+		String version = oclClient.getUrlVersion(url, subscriptionToken);
+		assertThat(version, is("v1.0"));
+	}
+
+	@Test
+	public void getExportUrl_shouldConstructCorrectExportUrlWhenVersionIsPassedToUrl() throws URIException {
+		String url = "https://api.openconceptlab.org/users/username/collections/collectionname/v1.0";
+		String collectionVersion = "v1.0";
+		GetMethod getMethod = oclClient.getExportUrl(url, collectionVersion);
+		assertThat(getMethod.getURI().toString(), is(url + "/export"));
+	}
+
+	@Test
+	public void getExportUrl_shouldConstructCorretExportUrlWhenVersionIsNotPassedToUrl() throws URIException {
+		String url = "https://api.openconceptlab.org/users/username/collections/collectionname/";
+		String collectionVersion = "v2.0";
+		GetMethod getMethod = oclClient.getExportUrl(url, collectionVersion);
+		assertThat(getMethod.getURI().toString(), is(url + "v2.0/export"));
+	}
+
+	@Test
+	public void removeLastUrlForwardSlashIfExist_shouldWorkAsExpected() {
+		String url1 = "https://api.openconceptlab.org/users/username/collections/collectionname/";
+		String url2 = "https://api.openconceptlab.org/users/username/collections/collectionname";
+		String newUrl1 = oclClient.removeLastUrlForwardSlashIfExist(url1);
+		String newUrl2 = oclClient.removeLastUrlForwardSlashIfExist(url2);

```

Comment:
The returned values being the same does not necessarily mean that the last forward slash has been removed. For instance, if some maintainer accidentally introduces a bug and the two method calls return `https://api.openconceptlab.org/users/username/collections/collectionname/`, this test will still pass.

Reply:
I'll add more check to cater for that

Reply:
Done